client/reader.js

```diff
@@ -59,93 +61,96 @@ const rl = readline.createInterface({
+                    datetime = moment(timeString, 'DD MMMM YYYY HH:mm:ss').unix();
+                }
+            } else if (line.match(/FLEX[:|] ?\d+-\d+-\d+ \d{2}:\d{2}:\d{2}/)) {
+                timeString = line.match(/\d+-\d+-\d+ \d{2}:\d{2}:\d{2}/)[0];
+                if (moment(timeString).isValid()) {
+                    datetime = moment(timeString).unix();
+                }
+            }
         }
-      }
-    }
-    if (line.match( /([ |]ALN[ |]|[ |]GPN[ |]|[ |]NUM[ |])/ )) {
-      message = line.match(/FLEX[:|].*[|\[][0-9 ]*[|\]] ?...[ |](.+)/)[1].trim();
-      if (line.match( /[ |][0-9]{4}\/[0-9]\/F\/.[ |]/ )) {
-        // message is fragmented, hold onto it for next line
-        frag[address] = message;
-        message = false;
-        trimMessage = '';
-      } else if (line.match( /[ |][0-9]{4}\/[0-9]\/C\/.[ |]/ )) {
-        // message is a completion of the last fragmented message
-        trimMessage = frag[address]+message;
-        delete frag[address];
-      } else if (line.match( /[ |][0-9]{4}\/[0-9]\/K\/.[ |]/ )) {
-        // message is a full message
-        trimMessage = message;
-      } else {
-        // message doesn't have the KFC flags, treat as full message
-        trimMessage = message;
-      }
-    }
-  } else if (line.match(/EAS[:|]/)) {
-     var decodedMeassage = SAME.decode(line); // Returns a list with [ Message, address ]
-      // Addresses are the following schema LLLL-ORG so for the exaple following the address is "KOAX-WXR" :  ZCZC-WXR-TOR-031109+0015-3650000-KOAX/NWS -
+        if (line.match(/([ |]ALN[ |]|[ |]GPN[ |]|[ |]NUM[ |])/)) {
+            message = line.match(/FLEX[:|].*[|\[][0-9 ]*[|\]] ?...[ |](.+)/)[1].trim();
+            if (line.match(/[ |][0-9]{4}\/[0-9]\/F\/.[ |]/)) {
+                // message is fragmented, hold onto it for next line
+                frag[address] = message;
+                message = false;
+                trimMessage = '';
+            } else if (line.match(/[ |][0-9]{4}\/[0-9]\/C\/.[ |]/)) {
+                // message is a completion of the last fragmented message
+                trimMessage = frag[address] + message;
+                delete frag[address];
+            } else if (line.match(/[ |][0-9]{4}\/[0-9]\/K\/.[ |]/)) {
+                // message is a full message
+                trimMessage = message;
+            } else {
+                // message doesn't have the KFC flags, treat as full message
+                trimMessage = message;
+            }
+        }
+    } else if (line.match(/(EAS[:|]|ZCZC-)/)) {                                                     // Adds EAS US/CA SAME Message Support          //Matches "EAS: ZCZC-ORG-EEE-PSSCCC+TTTT-JJJHHMM-CALL/FM -" OR "ZCZC-ORG-EEE-PSSCCC+TTTT-JJJHHMM-CALL/FM -" This allows future proofing or alternative feeding
+        var decodedMeassage = SAME.decode(line, EASOpts.excludeEvents, EASOpts.includeFIPS);          // Returns a object with all the info
```

Comment:
Minor thing before merge, can we fix typo on this var name 😁 

`decodedMeassage` -> `decodedMessage`

Reply:
😆 Nice catch @davidmckenzie lol

Was fixed by [I Learnt a book 925d8d4](https://github.com/pagermon/pagermon/pull/435/commits/925d8d46f0f9bec5b47d567238029959a1655b10)